addons/website/models/website.py

```diff
@@ -675,8 +675,98 @@ def configurator_apply(self, **kwargs):
+                html_string = ''.join(opening_tags) + '$0' + ''.join(closing_tags[::-1])
+
+                return html_string
+
+            def _generate_unique_hash(element_text, element_tag, element_attrib):
+                attrib_string = ','.join(f'{key}={value}' for key, value in sorted(element_attrib.items()))
+                combined_string = f'{element_text}|{element_tag}|{attrib_string}'
+                hash_object = hashlib.sha256(combined_string.encode('utf-8'))
+                return hash_object.hexdigest()
+
+            # Function to replace specific tags with placeholders
+            def _process_element(element):
+                # Handle highlights SVG.
+                if element.tag == 'span' and len(element) == 1 and element[0].tag == 'span' and len(element[0]) == 1 and element[0][0].tag == 'svg':
+                    # Extract the text from the inner <span>
+                    inner_text = element[0].text or ''
+                    element[0].text = '$0'
+                    # Store the outer HTML of the current element to reapply after chatGPT.
+                    outer_html_to_reapply = html.tostring(element, encoding='unicode', method='html')
+                    hash_value = _generate_unique_hash(inner_text, element.tag, element.attrib)
+                    html_storage[hash_value] = outer_html_to_reapply
+                    # Replace the content with the placeholder
+                    placeholder = f' ![{inner_text}](svg_{hash_value}) '
+                    element[0].text = ''
+                    element.text = placeholder
+                # Handle the maximum of inline tags (no nested tags for markdown syntax)
+                elif element.tag in {'u', 'em', 'strong', 's', 'br', 'font', 'b', 'small'} and len(element) == 0:
+                    if element.tag == 'br':
+                        placeholder = ' ![br](br) '
+                        element.tag = 'text'
+                    elif element.tag == 'font':
+                        hash_value = _generate_unique_hash(element.text, element.tag, element.attrib)
+                        placeholder = f'![{element.text}]({element.tag}_{hash_value})'
+                        fonts_attributes[hash_value] = element.attrib
+                    else:
+                        placeholder = f'![{element.text}]({element.tag})'
+                    element.text = placeholder
+
+            # Looking for one or several wrapping tags for all the HTML
+            # e.g. <strong><em>text ...</em><strong>
+            # keep in memory the wrapping tags to reapply after chatGPT.
+            wrapping_html = []
+            for element in tree.iter():
+                wrapping_html.append({"tag": element.tag, "attr": element.attrib})
+                if len(element) != 1 or element.text is not None:
+                    break
+
+            # Traverse the tree and process each element
+            for element in tree.iter():
+                _process_element(element)
```

Comment:
The `_process_element` function is useless: just add what it does directly inside this `for` loop

Reply:
agree